src/renderers/dom/shared/utils/isCustomComponent.js

```diff
@@ -11,8 +11,17 @@
 
 'use strict';
 
-function isCustomComponent(tagName, props) {
-  return tagName.indexOf('-') >= 0 || props.is != null;
+var DOMNamespaces = require('DOMNamespaces');
+var HTML_NAMESPACE = DOMNamespaces.Namespaces.html;
+
+function isCustomComponent(tagName, props, namespace) {
+  if (tagName.indexOf('-') >= 0 || props.is != null) {
+    // TODO: We always have a namespace with fiber. Drop the first
+    // check when Stack is removed.
+    return namespace == null || namespace === HTML_NAMESPACE;
```

Comment:
@gaearon I moved the namespace check into this module. Stack validates properties _before_ the namespace is assigned to the element, so we don't know if the element is SVG or HTML during initial validations. `namespace == null` can go away when stack is removed. This also preserves the custom element behavior for 15.x, if that matters.